### PR TITLE
Completely renewed pug interpolations

### PR DESCRIPTION
**Pull request by Patcher56**
using vscode syntax highlighting injections now
using original pug syntax with specific injections for vue

could delete whole pug implementation -> only use injections anymore
This should fix every pug syntax highlighting problem with vetur, except the one regarding interpolations and directives (such as #1154)

The advantages are:
- Much less code (cleaner code)
- easier to maintain
- When original pug syntax highlighting updates, this automatically gets updated too
- No own language for vue pug syntax highlighting (injected in the vue language)

This can be done with HTML too, should I do that @octref ?

Also added a commit to support new [v-slot](https://vuejs.org/v2/guide/components-slots.html#Named-Slots-Shorthand) syntax in 5eba764.

### CODE REVIEW

**Comment by Patcher56** [comment]
This also fixes #1139

File: syntaxes/vue.tmLanguage.json [hunk 1]
```diff
@@ -76,17 +76,9 @@
...
-        "1": {
-          "name": "punctuation.definition.tag.begin.html"
-        },
-        "2": {
-          "name": "entity.name.tag.template.html"
-        }
-      },
-      "end": "(</)(template)(>)",
-      "endCaptures": {
+      "begin": "(?:^\\s+)?(<)((?i:template))\\b(?=[^>]*lang=[\\\"'](?:pug|jade)[\\\"'])",
```
Comment (octref) [inline syntaxes/vue.tmLanguage.json]: Not quite sure what is this change for — do you mind explaining your reasoning here?

File: syntaxes/vue.tmLanguage.json [hunk 2]
```diff
@@ -102,17 +94,20 @@
...
-          "contentName": "text.pug.vue-pug",
+          "contentName": "source.vue.pug",
           "begin": "(>)",
           "beginCaptures": {
             "1": {
               "name": "punctuation.definition.tag.end.html"
             }
           },
-          "end": "(?=</template>)",
+          "end": "(?=</(?i:template))",
```
Comment (octref) [inline syntaxes/vue.tmLanguage.json]: Why are we dropping the `>`?

**Comment by octref** [comment]
Thanks a lot for the change! Reducing code duplication is always good.

> HTML too

Let's do another PR for that once this one gets merged in and people aren't reporting much issues.

File: syntaxes/vue.tmLanguage.json (hunk 1 above)
Comment (Patcher56) [reply syntaxes/vue.tmLanguage.json]: Sorry, those two are some leftovers due to some tests before. Reverted it in the last commit.

File: syntaxes/vue.tmLanguage.json (hunk 2 above)
Comment (Patcher56) [reply syntaxes/vue.tmLanguage.json]: See comment above.

File: syntaxes/vue-generated.json [hunk 3]
```diff
@@ -162,17 +162,9 @@
...
-      "begin": "(<)(template)\\b(?=[^>]*lang=('jade'|\"jade\"|'pug'|\"pug\"))(?![^/>]*/>\\s*$)",
-      "beginCaptures": {
-        "1": {
-          "name": "punctuation.definition.tag.begin.html"
-        },
-        "2": {
-          "name": "entity.name.tag.template.html"
-        }
-      },
+      "begin": "(<)(template)\\b(?=[^>]*lang=[\\\"'](?:pug|jade)[\\\"'])(?![^/>]*/>\\s*$)",
```
Comment (octref) [inline syntaxes/vue-generated.json]: I'd rather not have cleverness here — this allows `'pug"` and `"pug'`, etc.

File: syntaxes/vue-generated.json [hunk 4]
```diff
@@ -162,17 +162,9 @@
...
       "end": "(</)(template)(>)",
-      "endCaptures": {
+      "captures": {
```
Comment (octref) [inline syntaxes/vue-generated.json]: Shouldn't this still be `endCaptures`?

- `</` — `"punctuation.definition.tag.begin.html"`
- `template`  — `"entity.name.tag.template.html"`
- `>` — `"punctuation.definition.tag.end.html"`

File: syntaxes/vue-generated.json [hunk 5]
```diff
@@ -162,17 +162,9 @@
       ]
     },
     {
-      "begin": "(<)(template)\\b(?=[^>]*lang=('jade'|\"jade\"|'pug'|\"pug\"))(?![^/>]*/>\\s*$)",
-      "beginCaptures": {
```
Comment (octref) [inline syntaxes/vue-generated.json]: Why dropping the `beginCaptures`?

File: syntaxes/vue.yaml [hunk 6]
```diff
@@ -42,24 +42,21 @@ patterns:
...
 
   # jade / pug
-  - begin: (<)(template)\b(?=[^>]*lang=('jade'|"jade"|'pug'|"pug"))(?![^/>]*/>\s*$)
-    beginCaptures:
-      '1': { name: punctuation.definition.tag.begin.html }
-      '2': { name: entity.name.tag.template.html }
+  - begin: (<)(template)\b(?=[^>]*lang=[\"'](?:pug|jade)[\"'])(?![^/>]*/>\s*$)
     end: (</)(template)(>)
-    endCaptures:
+    captures:
```
Comment (octref) [inline syntaxes/vue.yaml]: See the comment above — not sure why the change to `captures` from `endCaptures` and dropping `beginCaptures`.

File: syntaxes/vue-generated.json (hunk 4 above)
Comment (Patcher56) [reply syntaxes/vue-generated.json]: As the `captures` keyword is just a shorthand for `beginCaptures` and `endCaptures` and we used the same definitions for `1` and `2` I thought it would be cleaner to just use `captures`.

If you want I can also change this back. The result is the same.

File: syntaxes/vue-generated.json (hunk 3 above)
Comment (Patcher56) [reply syntaxes/vue-generated.json]: Oh, you're right! I will change that asap. Sometimes the cleaner way still is not the best way...

File: syntaxes/vue-generated.json (hunk 4 above)
Comment (octref) [reply syntaxes/vue-generated.json]: I see, just to understand why you made the change.

Yeah I'd prefer them separate. Also with this you don't need to convert the pug/jade part to `(?:)`, although I would be surprised that had any effect since it's enclosed in an `(?=)` lookahead.

**Comment by octref** [comment]
LGTM 👍 Thanks a lot!
